imports/api/users/methods.js

```diff
@@ -103,6 +103,18 @@ if (Meteor.isClient) {
   });
 }
 
+export function mergeLastSeen(user, topicId, destinationId, seenType) {
+  const lastSeens = user.lastSeens();
+  if (!seenType) seenType = Meteor.users.SEEN_BY.EYES;
```

Comment:
Ez a seenType-os dolog, hogy EYES vagy NOTi, es az egyik implikalja a masikat, ez kicsit komplex. En irtem, de szeretnek elfeledkezni rola.
Vajon lehet e ezt a logikat kitenni egy fuggvenybe, es mindenhol azt hasznalni. Es a fv egy fv-t kap, hogy ezt a dolgot csinald meg rendesen minden noti tipusra.
Nem tudom hogy lehet e ilyet, meg nem neztem bele a kodba, csak hangosan gondolkodom, hogy hatha lehet ezt a komplexitast izolalni egy kulon fuggvenybe....

Reply:
Ezt az eszrevtelemet lehet hogy visszavonom. :)
Az zavart meg kicsit, hogy miert kell a lehetoseg hogy bekerni, hogy melyik seenType-ra akarom megcsinalni.
A koltoztetes nem azt jelenti hogy mindkettot koltoztetni kell?
Elkepzelhetonek tunik olyan eset, hogy a noti kuldes koltozik, de a szemmel latast nem akarjuk koltoztetni?

Reply:
Igen, ez itt elég bonyolultan lett leírva, keresem is még az egyszerűbb megoldást. Mind a kettőt költöztetjük, de különbözőek lehetnek a notis meg a byeyes-os lastseenek, és ezért a modifier is.  Mert ha notit már kapott, attól még lehet, hogy a szemmel látósat nagyon rég ellenőrizte, és akkor notit ugyan nem küldünk, de nem akarjuk a karikát is levenni.

Reply:
Igen, pont igy gondolom en is. Hogy a ketto teljesen fuggetlen egymastol. Es egzmastol fuggetlenul, de ugyanugy kell koltoztetni.
(a for ciklusnak vegig kell mennie mindketton, szerintem ez nem opcionalis)

Reply:
Ha pedig ugy is meg kell csinalni mindketton, akkor a for ciklusnal jobban kommunikalja ez egy [BY_EYES, BY_NOTI].forEach(...
a for ciklust, akkor szoktuk hasznalni, ha az index szamit.
Ott ahol en hasznaltam, az volt vele a cel, hogy az egyik sorrendben erosebb mint a masik, amiatt lett a for ciklus.
Es aztan elso ranezesre azt hittem itt is szamit a sorrend.

Reply:
Igy talan meg olvashatobb lenne a koltoztetes:
_.each(Meteor.users.SEEN_BY, (seenBy) => { 
.... megcsinalni a teendot